README.md

```diff
@@ -0,0 +1,376 @@
+      "displayName": "Temperature",
+      "name": "temperature",
+      "schema": "double"
+    },
+    {
+      "@type": [
+        "Telemetry",
+        "SemanticType/Humidity"
+      ],
+      "description": "Current humidity on the device",
+      "displayName": "Humidity",
+      "name": "humidity",
+      "schema": "double"
+    },
+    {
+      "@type": [
+        "Telemetry",
+        "SemanticType/Pressure"
+      ],
+      "description": "Current Pressure on the device",
+      "displayName": "Pressure",
+      "name": "pressure",
+      "schema": "double"
+    },
+    {
+      "@type": "Command",
+      "description": "Get a report with summary values from the past N days.",
+      "name": "GetClimateSummary",
+      "commandType": "synchronous",
+      "request": {
+        "name": "numberOfDays",
+        "schema": "integer"
+      },
+      "response": {
+        "name": "climateSummaryResponse",
+        "schema": {
+          "@type": "Object",
+          "fields": [
+            {
+              "name": "SummaryText",
+              "schema": "string"
+            },
+            {
+              "name": "highTemp",
+              "schema": "double"
+            },
+            {
+              "name": "lowTemp",
+              "schema": "double"
+            },
+            {
+              "name": "avgTemp",
+              "schema": "double"
+            },
+            {
+              "name": "numberOfSamples",
+              "schema": "float"
+            }
+          ]
+        }
+      }
+    }
+  ]  
+}
+```
+
+</details>
+
+The WeatherStation device implement the next interfaces:
+
+<details>
+
+<summary>WeatherStation model</summary>
+
+```json
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:azsamples:iotdd:WeatherStation;1",
+  "@type": "Interface",
+  "displayName": "Sample Weather Station with interior and exterior sensors",
+  "contents": [
+    {
+      "@type": "Component",
+      "schema": "dtmi:azsamples:iotdd:ClimateSensor;1",
+      "name": "interior"
+    },
+    {
+      "@type": "Component",
+      "schema": "dtmi:azsamples:iotdd:ClimateSensor;1",
+      "name": "exterior"
+    },
+    {
+      "@type": "Component",
+      "schema": "dtmi:azure:DeviceManagement:DeviceInformation;1",
+      "name": "deviceInfo"
+    },
+    {
+      "@type": "Component",
+      "schema": "dtmi:azure:Client:SDKInformation;1",
+      "name": "sdkInfo"
+    },
+    {
+      "@type": "Component",
+      "schema": "dtmi:azsamples:iotdd:Diagnostics;1",
+      "name": "diag"
+    }
+  ]
+}
+```
+
+</details>
+
+### CloudAplication
+
+This is an application that connects to IoT Hub using the service SDK offering the next features:
+
+- List Devices and Status
+- For Each Device:
+  - Determine if the device announces a ModelId
+  - Resolve the Model contents from the ModelId
+    - If it's a known model, show an ad-hoc UI
+  - For Models without an ad-hov UI
+    - Show a dynamically generated UI
```

Comment:
>Show a dynamically generated UI [](start = 6, length = 31)

Does this mean just list out the information for about the model

Reply:
this means to show a UI like IoT Explorer. There will be limitations wrt complex types, etc.. lets start to support primitive types and commands with simple in/out params. I will add more details to the doc.